cedar-policy-core/src/parser/err.rs

```diff
@@ -803,14 +806,25 @@ impl<T: Into<ParseError>> From<T> for ParseErrors {
 }
 
 impl From<Vec<ParseError>> for ParseErrors {
+    /// Convert a `Vec<ParseError> to a `ParseErrors`, inserting the `Unknown`
+    /// error if necessary.
     fn from(errs: Vec<ParseError>) -> Self {
-        ParseErrors(errs)
+        match errs.split_first() {
+            None => ParseErrors::singleton(ParseError::ToAST(ToASTError::new(
+                ToASTErrorKind::Unknown,
+                Loc::new(0, Arc::from("")),
+            ))),
+            Some((first, rest)) => ParseErrors::new(first.clone(), rest.iter().cloned()),
+        }
```

Comment:
any way to avoid cloning all the errors?  maybe `Vec::drain()`?  (minor nit though, as performance on the error path isn't as important)

Reply:
Switched to using `from_iter()` , which includes an extra collect, but no clone.

(But note that this function will be deleted in a future PR.)